c++/src/kj/BUILD.bazel

```diff
@@ -116,6 +116,8 @@ cc_library(
         "//conditions:default": ["async-unix.h"],
     }),
     include_prefix = "kj",
+    # Disable header parsing based on async-inl.h
+    features = ["-parse_headers"],
```

Comment:
Disabling the feature here means that Bazel will not parse the headers of kj-async. This is the cleanest way to get header parsing to work given the `#error "Do not include this directly; include kj/async.h."` directive in `async-inl.h`. Alternatively we could add a define to allow disabling that check while still including async.h but this is easier.